### ISSUE TITLE
Blocks sometimes fall through each other/the ground

### ISSUE THREAD

**Issue by Nicholas-Steel** [renaudbedard/fez-1.12-issues#145]
Demonstration: https://youtu.be/bpi7055i27I (Watch from 1:30)

The problem makes it harder than it should be to solve the puzzle. I'm not entirely sure what is going on but it looks like blocks lose detection of the ground briefly when rotating the screen, causing them to fall through the ground/other blocks if their stacked up?

Sometimes you can push a block on a lower level and have it move a block in tandem that is diagonally up from it (Maybe it was just barely resting on top of the lower block?) This isn't demonstrated in the video (I did try to reproduce it).

**Comment by renaudbedard** [comment]
I realize that this is a real issue, but it's been happening since day one and at this point I'm not touching crate physics in FEZ, it's a pandora's box that could open up a dozen other bugs.

Thanks for reporting however!

**Comment by Nicholas-Steel** [comment]
Watch the video from 1:30, while you may not fix it I think it is worse than it ever was in the past? So something seems to have exasperated the issues. I'll test it in Single Threaded mode, that might improve the behaviour.

I know on Xbox 360 there were some issues, though it wasn't as bad as this (If you exclude the ability to deadlock the game by somehow lifting a block while you're not standing on ground properly).

**Comment by Nicholas-Steel** [comment]
Refer to https://github.com/renaudbedard/fez-1.12-issues/issues/145#issuecomment-244939988

**Comment by renaudbedard** [comment]
Got it, I'll look into it to see if something new broke.

**Comment by Nicholas-Steel** [comment]
Just to note, the issues take place as the screen is rotated. I've updated the above post to clarify this.

If you can get block physics to be as good as they are on the Xbox 360 port, and maybe get the visualizations effects during the ending sequence to display without duplicate frames/stutter (at 60hz) I think you'll have fixed enough for this to be the definitive version of the game (It's incredibly close to being the best version).

**Comment by renaudbedard** [comment]
@Nicholas-Steel : If you ever feel like giving this puzzle another go, I fixed a bunch of issues that it had in the "experimental" branch. Let me know if you can give it another go!

**Comment by Nicholas-Steel** [comment]
I did manage to get past it, I 100%'d the game just before I unsubscribed from the other bug report. I can of course mess with it in my completed save game file. I'll look in to it later if I remember.

**Comment by Nicholas-Steel** [comment]
If you have a stack of 2 blocks (one on top of the other) in front of terrain with the top block on both a block and the terrain (So you are looking at the stack from a perspective that shows the top block as sitting on a block as well as on the terrain), rotating the screen so that you view the blocks in front of the terrain and then rotating the screen again will cause the top block to shift on to the terrain behind the stack. (It should stay on the stack if it is placed on the stack!)

If you have a stack of 3 blocks in front of terrain with the middle block on both the bottom block and looking like it is also sitting on terrain (similar setup to the first issue but this time with a 3rd block that is only supported by the 2nd block), rotating the screen will cause the middle block to shift on to the terrain behind the stack and the blocks that were supported by it will bug out/persist in the air at their original location and be visible at some angles and invisible with buggy properties at other angles (Alters how your character functions when interacting with it/hanging from it).

Blocks can fall through the floor/other blocks randomly (I am unsure what specific circumstances cause this, it feels random but I'm fairly confident it isn't actually random). This does occur in the video demonstration I linked to in the first post. If a block sinks in to another block they will literally occupy the exact same space, thankfully the game only lets you push one block at a time so it is possible to separate them.

I don't know if you already fixed any of these issues. This is a clearer explanation of the issues I tried to report 14 days ago.
